### PR TITLE
Log errors from probe

### PR DESCRIPTION
**Pull request by jkroepke**
This PR adds a flag `--log.prober` which can be one of: [debug, info, warn, error, none] to investigate probe failures port-mortem. this flag defaults to none.

At the moment, this analyse problems is only possible via Web UI with an large enough history. Access the unprotected web ui can be difficult and the probe history is only available until restart.

### CODE REVIEW

File: main.go
Comment (SuperQ) [inline main.go:53]:
```suggestion
	logProbeErrors = kingpin.Flag("log.probe-errors", "Log error from probe requests").Default().Bool()
```

**Review by SuperQ — CHANGES_REQUESTED** [review]

**Comment by jkroepke** [comment]
@electron0zero Could you please also take a look here?

File: prober/handler.go
Comment (electron0zero) [inline prober/handler.go:185]: this will only log errors from the checks? maybe we can make this configurable where users can configure the level they want to log? 

this would make this flag bit generic and more useful. we can rename the flag to something like `--log.prober=info`, and you can configure it to `--log.prober=error` to log only error, and replicate what we have in current state.

IMO this feature will be useful to users who are running BBE with central log collection in place.

we can set it to `error` by default or disable it by default, I am fine with both options.

cc @SuperQ what do you think?

Comment (jkroepke) [reply prober/handler.go:185]: What if `--log.prober=info`, but  `--log.level=warn` if set? 

Should info logs still visible on stdout?

Comment (jkroepke) [reply prober/handler.go]: @electron0zero @SuperQ how we can proceed here?

**Review by electron0zero — APPROVED** [review]
lgtm, will wait for @roidelapluie or @mem to take a look and approve :)

**Comment by jkroepke** [comment]
@electron0zero @roidelapluie or @mem How we can proceed here?

**Comment by electron0zero** [comment]
assuming that there are no comments or feedback from other maintainers, I am going to go ahead and merge it.

**Pull request merged**
